src/GitVersionCore/Helpers/IFileSystem.cs

```diff
@@ -13,5 +15,10 @@ public interface IFileSystem
         void WriteAllText(string file, string fileContents);
         IEnumerable<string> DirectoryGetFiles(string directory, string searchPattern, SearchOption searchOption);
         Stream OpenWrite(string path);
+        Stream OpenRead(string path);
+        void CreateDirectory(string path);
+        long GetLastDirectoryWrite(string path);
+        string TreeWalkForDotGitDir(string directory);
```

Comment:
Nice. I've replaced all usage of `TreeWalkForDotGitDir()` with `Repository.Discover()`.